File: app/src/main/java/com/masterclass/FoodLove/Activity/ShowDetailActivity.java
package com.masterclass.FoodLove.Activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.masterclass.FoodLove.Domain.FoodDomain;
import com.masterclass.FoodLove.Helper.ManagementCart;
import com.masterclass.FoodLove.R;

public class ShowDetailActivity extends AppCompatActivity {

    private TextView addToCartBtn;
    private TextView titleTxt,feeTxt,descTxt,numberOrderTxt;
    private ImageView plusBtn, minusBtn, picFood;
    private FoodDomain object;
    int numberOrder = 1;
    private ManagementCart managementCart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_detail);

        managementCart = new ManagementCart(this);

        initView();
        getBundle();
    
    }

    private void getBundle() {
        object = (FoodDomain) getIntent().getSerializableExtra("object");

        int drawableResourceId = this.getResources().getIdentifier(object.getPic(),"drawable",this.getPackageName());
        Glide.with(this)
                .load(drawableResourceId)
                .into(picFood);

        titleTxt.setText(object.getTitle());
        feeTxt.setText("€"+object.getFee());
        descTxt.setText(object.getDesc());
        numberOrderTxt.setText(String.valueOf(numberOrder));

        plusBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                numberOrder++;
                numberOrderTxt.setText(String.valueOf(numberOrder));
            }
        });

        minusBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (numberOrder > 1){
                    numberOrder--;
                    numberOrderTxt.setText(String.valueOf(numberOrder));
                }
            }
        });

        addToCartBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                object.setNumberInCart(numberOrder);
                managementCart.insertFood(object);
            }
        });
    }

    private void initView() {
        addToCartBtn = findViewById(R.id.addToCartBtn);
        titleTxt =  findViewById(R.id.titleTxt);
        feeTxt = findViewById(R.id.fee);
        descTxt = findViewById(R.id.descTxt);
        numberOrderTxt = findViewById(R.id.numberItemTxt);
        plusBtn =  findViewById(R.id.plusBtn);
        minusBtn = findViewById(R.id.minusBtn);
    }
}

File: app/src/main/java/com/masterclass/FoodLove/Activity/ShowDetailActivity.kt
package com.masterclass.FoodLove.Activity

import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.masterclass.FoodLove.Domain.FoodDomain
import com.masterclass.FoodLove.Helper.ManagementCart
import com.masterclass.FoodLove.R

class ShowDetailActivity : AppCompatActivity() {
    private var addToCartBtn: TextView? = null
    private var titleTxt: TextView? = null
    private var feeTxt: TextView? = null
    private var descriptionTxt: TextView? = null
    private var numberOrderTxt: TextView? = null
    private var plusBtn: ImageView? = null
    private var minusBtn: ImageView? = null
    private var picFood: ImageView? = null
    private var `object`: FoodDomain? = null
    var numberOrder = 1
    private var managementCart: ManagementCart? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_detail)
        managementCart = ManagementCart(this)
        initView()
        bundle
    }

    private val bundle: Unit
        private get() {
            `object` = intent.getSerializableExtra("object") as FoodDomain?
            val drawableRessourceId =
                this.resources.getIdentifier(`object`!!.pic, "drawable", this.packageName)
            Glide.with(this)
                .load(drawableRessourceId)
                .into(picFood!!)
            titleTxt!!.text = `object`!!.title
            feeTxt!!.text = "$" + `object`!!.fee
            descriptionTxt!!.text = `object`!!.desc
            numberOrderTxt!!.text = numberOrder.toString()
            plusBtn!!.setOnClickListener {
                numberOrder = numberOrder + 1
                numberOrderTxt!!.text = numberOrder.toString()
            }
            minusBtn!!.setOnClickListener {
                if (numberOrder > 1) {
                    numberOrder = numberOrder - 1
                }
                numberOrderTxt!!.text = numberOrder.toString()
            }
            addToCartBtn!!.setOnClickListener {
                `object`!!.numberInCart = numberOrder
                managementCart!!.insertFood(`object`!!)
            }
        }

    private fun initView() {
        addToCartBtn = findViewById(R.id.addToCartBtn)
        titleTxt = findViewById(R.id.titleTxt)
        feeTxt = findViewById(R.id.priceTxt)
        descriptionTxt = findViewById(R.id.descriptionTxt)
        numberOrderTxt = findViewById(R.id.numberOrderTxt)
        plusBtn = findViewById(R.id.plusBtn)
        minusBtn = findViewById(R.id.minusBtn)
        picFood = findViewById(R.id.picFood)
    }
}